python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -87,3 +161,75 @@ cpdef read_parquet(filepath_or_buffer, columns=None, row_group=None,
         df.index.name = new_index_name
 
     return df
+
+cpdef write_parquet(
+        Table table,
+        path,
+        index=None,
+        compression=None,
+        statistics="ROWGROUP"):
+    """
+    Cython function to call into libcudf API, see `write_parquet`.
+
+    See Also
+    --------
+    cudf.io.parquet.write_parquet
+    """
+
+    # Create the write options
+    cdef string filepath = <string>str(path).encode()
+    cdef sink_info sink = sink_info(filepath)
+    cdef unique_ptr[table_metadata] tbl_meta = \
+        make_unique[table_metadata]()
+
+    cdef vector[string] column_names
+    cdef map[string, string] user_data
+    cdef table_view tv = table.data_view()
+
+    if index is not False:
+        tv = table.view()
+        if isinstance(table._index, cudf.core.multiindex.MultiIndex):
+            for idx_name in table._index.names:
+                column_names.push_back(str.encode(idx_name))
+        else:
+            column_names.push_back(str.encode(table._index.name))
```

Comment:
@shwina is there a way to iterate over the names of the index levels regardless of whether it's an index or MultiIndex?

Reply:
@jdye64 connected with @shwina offline and the best path forward here is to fix the `names` property of the Index class to properly exist for all of the Index dtypes instead of just `MultiIndex` and `CategoricalIndex` currently. Should be super straightforward.

Reply:
@shwina @kkraus14 I know a multiindex is the only index implementation with the .names attribute hence why I wrote it this way, but yeah if their is a way to iterate over them the same way I'm happy to do that as the duplicative sections didn't make me happy to place there. Let me know.

Reply:
Ignore my last comment, had a stale github window open and your previous response wasn't there when I wrote my comment. Sorry. Will follow your advice.